src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -52,11 +52,12 @@ private void CloseInternal()
             _exception = s_disposedSentinel;
             CloseContext();
 
-            // Ensure a Read operation is not in progress,
-            // block potential reads since SslStream is disposing.
-            // This leaves the _nestedRead = 1, but that's ok, since
-            // subsequent Reads first check if the context is still available.
-            if (Interlocked.CompareExchange(ref _nestedRead, 1, 0) == 0)
+            // Ensure a Read or Auth operation is not in progress,
+            // block potential future read and auth operations since SslStream is disposing.
+            // This leaves the _nestedRead = 1 and _nestedAuth = 1, but that's ok, since
+            // subsequent operations check the _exception sentinel first
+            if (Interlocked.CompareExchange(ref _nestedRead, 1, 0) == 0 &&
+                Interlocked.CompareExchange(ref _nestedAuth, 1, 0) == 0)
```

Comment:
These are only ever 0 or 1, right?  Can these just be:
```suggestion
            if (Interlocked.Exchange(ref _nestedRead, 1) == 0 &&
                Interlocked.Exchange(ref _nestedAuth, 1) == 0)
```
?

Reply:
It took me a while to prove to myself that these two are equivalent